### PR TITLE
Adding Transactions and BillingSubscription to the BillingRP spec

### PR DESCRIPTION
**Pull request by asarkar84**
Adding Transactions and BillingSubscription to the BillingRP spec

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by asarkar84** [comment]
We have raised this pull request to Get ARM review and Sign off for updating the ARM manifest in PROD. There was already an ARM approved PR: https://github.com/Azure/azure-rest-api-specs/pull/3900 

I have taken all files from the above PR and added only Transactions and BillingSubscriptions.
Please review these paths only which are added newly in this PR:
"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/transactions"
"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingProfiles/{billingProfileId}/transactions"
"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingSubscriptions"
"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingProfile/{billingProfileId}/billingSubscriptions"
"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/projects/{projectId}/billingSubscriptions"
"/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/projects/{projectId}/billingSubscriptions/{billingSubscriptionId}"

**Comment by asarkar84** [comment]
Please do not merge this PR.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json
Comment (RyanBensonMSFT) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-08-01-preview/billing.json]: Product group agreed to fix this across all their API's before removing -preview.

**Review by ryansbenson — APPROVED** [review]
Approved

**Comment by jianghaolu** [comment]
@asarkar84 @marstr 
Yes please address my comments in #3900. thanks!

Also please add [Merge on {date}] in front the PR title to prevent accidental merge.

**Review by marstr — APPROVED** [review]
Approving by transitivity as @jianghaolu has given former approval.

**Comment by marstr** [comment]
@asarkar84 when would you like this PR to be merged?

**Comment by marstr** [comment]
Howdy @asarkar84. I'm going to close this as stale. When you want it merged, please feel free to ping me to have it re-opened or to submit a new PR.

**Pull request closed without merging**
